components/brave_federated/data_stores/data_store.h

```diff
@@ -30,10 +32,15 @@ class DataStore {
             const std::string& task_name,
             int max_number_of_records,
             int max_retention_days);
-  bool DeleteLogs();
+
+  typedef base::flat_map<int, std::vector<mojom::Covariate>> TrainingData;
```

Comment:
this is the reason of crashes - use after free. Need to store mojom::CovariatePtr here